runtime/src/main/java/org/corfudb/runtime/collections/CorfuStore.java

```diff
@@ -0,0 +1,130 @@
+package org.corfudb.runtime.collections;
+
+import com.google.protobuf.Message;
+
+import java.lang.reflect.InvocationTargetException;
+import java.util.Collection;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+import org.corfudb.protocols.wireprotocol.Token;
+import org.corfudb.runtime.CorfuRuntime;
+import org.corfudb.runtime.CorfuStoreMetadata.TableName;
+import org.corfudb.runtime.CorfuStoreMetadata.Timestamp;
+
+/**
+ * Created by zlokhandwala on 2019-08-02.
+ */
+public class CorfuStore {
+
+    private final CorfuRuntime runtime;
+
+    /**
+     * Creates a new CorfuStore.
+     *
+     * @param runtime Connected instance of the Corfu Runtime.
+     */
+    @Nonnull
+    public CorfuStore(@Nonnull final CorfuRuntime runtime) {
+        this.runtime = runtime;
+    }
+
+    /**
+     * Fetches the latest logical timestamp in Corfu's distributed log.
+     *
+     * @return Timestamp.
+     */
+    @Nonnull
+    public Timestamp getTimestamp() {
+        Token token = runtime.getSequencerView().query().getToken();
+        return Timestamp.newBuilder()
+                .setEpoch(token.getEpoch())
+                .setSequence(token.getSequence())
+                .build();
+    }
+
+    /**
+     * Creates and registers a table.
+     * A table needs to be registered before it is used.
+     *
+     * @param namespace    Namespace of the table.
+     * @param tableName    Table name.
+     * @param kClass       Class of the Key Model.
+     * @param vClass       Class of the Value Model.
```

Comment:
nit - update header to include meta class

Reply:
done. thanks

Reply:
done.